loader/include/Geode/ui/TextAreaV2.hpp

```diff
@@ -0,0 +1,241 @@
+#pragma once
+
+#include <cocos2d.h>
+#include "PaddingNode.hpp"
+
+namespace geode {
+    class GEODE_DLL TextAreaV2 : public PaddingNode {
+    public:
+        enum class WrappingMode {
+            // Doesn't wrap the text and completely ignores bounds
+            NoWrap,
+            // Wraps the text on the last special character (spaces included) before the width is exceeded
+            WordWrap,
+            // Wraps the text on the last space before the width is exceeded
+            SpaceWrap,
+            // Wraps the text on the exact character that exceeds the width
+            CutoffWrap
+        };
+
+        enum class Alignment {
+            // Aligns the text to the left
+            Left,
+            // Aligns the text in the center
+            Center,
+            // Aligns the text to the right
+            Right
+        };
+
+        struct Line {
+            std::string text;
+            std::string overflow;
+            cocos2d::CCLabelBMFont* label;
+            int lineNumber;
+            float currentHeight;
+            bool isLastLine;
+        };
+
+        static TextAreaV2* create(const std::string& text, const std::string& font, const float scale = 1, const float width = -1, const float height = -1, const bool deferUpdates = false);
+
+        /**
+        * Sets the font of the text area
+        */
+        void setFont(const std::string& font);
+        /**
+        * Gets the font of the text area
+        */
+        std::string getFont() const;
+        /**
+        * Sets the text of the text area
+        */
+        void setText(const std::string& text);
+        /**
+        * Gets the text of the text area
+        */
+        std::string getText() const;
+        /**
+        * Sets the color of the text in the text area
+        */
+        void setTextColor(const cocos2d::ccColor4B& color);
+        /**
+        * Gets the color of the text in the text area
+        */
+        cocos2d::ccColor4B getTextColor() const;
+        /**
+        * Sets horizontal the alignment of the text in the text area
+        */
+        void setAlignment(const Alignment alignment);
```

Comment:
This is fine because it's not a ref, though the `const` here is also utterly redundant